third_party/move/move-binary-format/src/builders.rs

```diff
@@ -0,0 +1,321 @@
+            || ModuleHandle { address, name },
+            (address, name),
+        )
+        .map(|idx| ModuleHandleIndex(idx as u16))
+    }
+
+    pub fn import_struct(
+        &mut self,
+        module: &CompiledModule,
+        idx: StructHandleIndex,
+    ) -> PartialVMResult<StructHandleIndex> {
+        let handle = module.struct_handle_at(idx);
+        let module_id = self.import_module(module, handle.module)?;
+        let name = self.import_identifier(module, handle.name)?;
+        let idx = get_or_add_impl(
+            &mut self.script.struct_handles,
+            &mut self.struct_pool,
+            || StructHandle {
+                module: module_id,
+                name,
+                abilities: handle.abilities,
+                type_parameters: handle.type_parameters.clone(),
+            },
+            (module_id, name),
+        )?;
+        Ok(StructHandleIndex(idx as u16))
+    }
+
+    pub fn import_signature_token(
+        &mut self,
+        module: &CompiledModule,
+        sig: &SignatureToken,
+    ) -> PartialVMResult<SignatureToken> {
+        use SignatureToken::*;
+        Ok(match sig {
+            U8 => U8,
+            U16 => U16,
+            U32 => U32,
+            U64 => U64,
+            U128 => U128,
+            U256 => U256,
+            Bool => Bool,
+            Address => Address,
+            Signer => Signer,
+            TypeParameter(i) => TypeParameter(*i),
+            Reference(ty) => Reference(Box::new(self.import_signature_token(module, ty)?)),
+            MutableReference(ty) => {
+                MutableReference(Box::new(self.import_signature_token(module, ty)?))
+            },
+            Vector(ty) => Vector(Box::new(self.import_signature_token(module, ty)?)),
+            Struct(idx) => Struct(self.import_struct(module, *idx)?),
+            StructInstantiation(idx, inst_tys) => StructInstantiation(
+                self.import_struct(module, *idx)?,
+                inst_tys
+                    .iter()
+                    .map(|sig| self.import_signature_token(module, sig))
+                    .collect::<PartialVMResult<Vec<_>>>()?,
+            ),
+        })
+    }
+
+    pub fn import_signatures(
+        &mut self,
+        module: &CompiledModule,
+        idx: SignatureIndex,
+    ) -> PartialVMResult<SignatureIndex> {
+        let sig = Signature(
+            module
+                .signature_at(idx)
+                .0
+                .iter()
+                .map(|sig| self.import_signature_token(module, sig))
+                .collect::<PartialVMResult<Vec<_>>>()?,
+        );
+        self.add_signature(sig)
+    }
+
+    pub fn add_signature(&mut self, signature: Signature) -> PartialVMResult<SignatureIndex> {
+        get_or_add(
+            &mut self.script.signatures,
+            &mut self.signature_pool,
+            signature,
+        )
+        .map(|idx| SignatureIndex(idx as u16))
+    }
+
+    pub fn import_call(
```

Comment:
Minor nit: I do find the name "import_call" a little bit confusing, in the sense that we're not really importing a call from another module, but rather functions and types that are needed by the call. Don't have a better name myself though.

Reply:
Renamed to `import_function_by_handle`